components/user-store/org.wso2.carbon.identity.user.store.configuration/src/main/java/org/wso2/carbon/identity/user/store/configuration/utils/DefaultUserStoreAttributeMappingParser.java

```diff
@@ -0,0 +1,160 @@
+
+import org.apache.axiom.om.OMAttribute;
+import org.apache.axiom.om.OMElement;
+import org.apache.axiom.om.impl.builder.StAXOMBuilder;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.user.store.configuration.model.UserStoreAttributeDO;
+import org.wso2.carbon.utils.CarbonUtils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import javax.xml.namespace.QName;
+import javax.xml.stream.XMLStreamException;
+
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.ATTRIBUTE_ID;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.CLAIM_CONFIG;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.CLAIM_URI;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.DIALECT;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.DIALECTS;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.DISPLAY_NAME;
+import static org.wso2.carbon.identity.user.store.configuration.utils.UserStoreConfigurationConstant.LOCAL_DIALECT_URL;
+
+/**
+ * Parser to read default user store mappings from claim-config.xml.
+ */
+public class DefaultUserStoreAttributeMappingParser {
+
+    private static Map<String, UserStoreAttributeDO> defaultAttributes;
+    private static final Log LOG = LogFactory.getLog(DefaultUserStoreAttributeMappingParser.class);
+
+    private DefaultUserStoreAttributeMappingParser() {
+
+        init();
+    }
+
+    private static final class ParserHolder {
+
+        static final DefaultUserStoreAttributeMappingParser PARSER = new DefaultUserStoreAttributeMappingParser();
+    }
+
+    public static DefaultUserStoreAttributeMappingParser getInstance() {
+
+        return ParserHolder.PARSER;
+    }
+
+    private static void init() {
+
+        File attributeMappingXml = new File(CarbonUtils.getCarbonConfigDirPath(), CLAIM_CONFIG);
+
+        if (attributeMappingXml.exists()) {
+            try (InputStream inStream = new FileInputStream(attributeMappingXml)) {
+                if (inStream == null) {
+                    String message = String.format("Claim-config configuration is not found at: %s/%s.",
+                            CarbonUtils.getCarbonConfigDirPath(), CLAIM_CONFIG);
+                    throw new FileNotFoundException(message);
+                }
+                buildDefaultAttributeMapping(inStream);
+            } catch (FileNotFoundException e) {
```

Comment:
I am bit confused here.. Why would we need to throw a file not found exception from line 84 and catch it here ? Exception handling is a cost. We don't need to do it unless we really need it

Reply:
Seems to be `inStream` is never null and what's caught from here is the exception thrown from the `FileInputStream` constructor.

Reply:
Yes it is true. I'm fixing it.

Reply:
fixed https://github.com/wso2/carbon-identity-framework/pull/3781